initialization/initialization_test.go

```diff
@@ -751,7 +748,7 @@ func TestValidateMetadata(t *testing.T) {
 		WithLogger(testLogger{t: t}),
 	)
 	r.ErrorAs(err, &errConfigMismatch)
-	r.Equal("NumUnits", errConfigMismatch.Param)
```

Comment:
Is increasing `NumUnits` now allowed? This would possibly require to look for a new VRF nonce.

Reply:
Do you mean that the new number of labels will invalidate the prior difficulty function?

Reply:
Yes; because the difficulty depends on `numUnits`.

If we would allow an increase of `numUnits` an adversary could look for a nonce with the smallest PoST allowed (numUnits = 1) and then when they found one that's beneficial for them increase their PoST size to the maximum size allowed.

This would essentially result in the same difficulty for all PoST sizes.

Reply:
I think currently it will just attempt to use a nonce potentially to be found invalid by peers. 

It seems that the solution is to just invalidate both `nonce` and `lastPosition` whenever the number of units is found to be larger than what's stored in the metadata. 
Anything I'm missing?

Reply:
We have no way of publishing a new nonce when a smesher increases their PoST. This is a ticket for after genesis: https://github.com/spacemeshos/post/issues/90

Increasing PoST size would require recalculating all labels already calculated to look for a new nonce that satisfies the increased difficulty. To avoid this overhead in PoST initialization cost we have to update `gpu-post` to not return the first nonce found to be below the difficulty threashold but the overall lowest nonce it finds in all labels calculated during initialization.

This is the only way we can guarantee that a smesher doesn't have to re-do all the work they already did.

Reply:
I assumed that it won't require to recalc labels, hence no code changes are needed. The whole point of doing PoW together with the labels is to amortize the work. If labels are already done, you can just continue to do PoW-only for higher positions.

This can indeed introduce a duplicated amount of PoW, and storing the lowest nonce will solve this, but I remembered that @tal-m originally [commented](https://github.com/spacemeshos/pm/issues/172#issuecomment-1245497516) against it. Reading it again, I think we just need to do some benchmarks. 

If other features besides properly amortized PoW are still missing for supporting a different nonce (in ATX publishing, etc.), I guess this is not relevant now. Are you suggesting to disallow the increase of `NumUnits`?